concert/experiments/imaging.py

```diff
@@ -60,3 +66,451 @@ def tomo_max_speed(frame_width, frame_rate):
+                                         self._vertical_motor.set_position(vertical_position))
+
+                    try:
+                        await self._camera.trigger()
+                    except:
+                        pass
+                    yield await self._camera.grab()
+        finally:
+            await self._finish_radios()
+
+
+class ContinuousSpiralTomography(ContinuousTomography, SpiralMixin):
+    """
+    Spiral Tomography
+
+    This implements a helical acquisition scheme, where the sample is translated perpendicular to the beam while the
+    sample is rotated and the projections are recorded.
+    """
+    vertical_velocity = Quantity(q.mm / q.s)
+
+    def __init__(self, walker, flat_motor, tomography_motor, vertical_motor, radio_position, flat_position,
+                 camera, start_position_vertical, sample_height, vertical_shift_per_tomogram,
+                 num_flats=200, num_darks=200, num_projections=3000, angular_range=180 * q.deg, start_angle=0 * q.deg,
+                 separate_scans=True):
+        ContinuousTomography.__init__(self, walker=walker, flat_motor=flat_motor, tomography_motor=tomography_motor,
+                                      radio_position=radio_position,
+                                      flat_position=flat_position,
+                                      camera=camera, num_flats=num_flats, num_darks=num_darks,
+                                      num_projections=num_projections, angular_range=angular_range,
+                                      start_angle=start_angle,
+                                      separate_scans=separate_scans)
+        SpiralMixin.__init__(self, vertical_motor=vertical_motor,
+                             vertical_shift_per_tomogram=vertical_shift_per_tomogram,
+                             sample_height=sample_height,
+                             start_position_vertical=start_position_vertical)
+
+    async def _get_vertical_velocity(self):
+        shift = await self.get_vertical_shift_per_tomogram()
+        fps = await self._camera.get_frame_rate()
+        num = await self.get_num_projections()
+
+        return shift * fps / num
+
+    async def _prepare_radios(self):
+        await asyncio.gather(
+            ContinuousTomography._prepare_radios(self),
+            self._vertical_motor.set_position(await self.get_start_position_vertical())
+        )
+
+    async def _finish_radios(self):
+        if self._finished:
+            return
+        await self._prepare_darks()
+        await self._tomography_motor.stop()
+        await self._vertical_motor.stop()
+        await self._vertical_motor['motion_velocity'].restore()
+        await asyncio.gather(
+            self._tomography_motor.set_position(await self.get_start_angle()),
+            self._vertical_motor.set_position(await self.get_start_position_vertical())
+        )
+        self._finished = True
+
+    async def _take_radios(self):
```

Comment:
Apart from the vertical motion, this is the same as in the continuous tomography case, should be also refactored.